src/routes/volume/VolumeBulkActions.js

```diff
@@ -114,15 +114,8 @@ function bulkActions({ selectedRows, engineImages, bulkDeleteVolume, showBulkEng
   const canUpgradeEngine = () => selectedRows.some(item => (item.state !== 'detached' && item.state !== 'attached'))
   const isFaulted = () => selectedRows.some(item => item.robustness === 'faulted')
   const isHasPVC = () => selectedRows.some(item => item.kubernetesStatus && item.kubernetesStatus.pvStatus && item.kubernetesStatus.pvStatus === 'Bound')
-  const isRestoring = () => selectedRows.some((selected) => {
-    if (selected.restoreStatus && selected.restoreStatus.length > 0) {
-      let flag = selected.restoreStatus.every((item) => {
-        return !item.isRestoring
-      })
-      return !flag
-    } else {
-      return false
-    }
+  const hasVolumeRestoring = () => selectedRows.some((selected) => {
+    return isRestoring(selected)
   })
   const isAutomaticallyUpgradeEngine = () => {
     return selectedRows.some((item) => {
```

Comment:
In the future we can do some more refactoring, by moving the single case into the helper.
We might want to just have a single function exported `canUpgradeEngineImage`
This refactor is not necessary for this PR, just something to think about.
example:
```
export func canUpgradeEngine(volume, image, defaultImage, autoUpgradeLimit) {
// we pass in the required parameters for determination, this way this can easily be unit tested and helper does not
// require knowledge of the surrounding context.
// image would be the desired image
// defaultImage would be string name of the image
// autoUpgradeLimit would be int number
}
```